src/rdiff-backup-delete

```diff
@@ -225,23 +226,45 @@ def _remove_increments(path, dry_run):
 
 def _rmtree(path, dry_run):
     """
-    Custom implementation of shutil.rmtree() to support deletion of symlink.
+    Custom implementation of shutil.rmtree() to handle permission errors,
+    fifo and symlink deletion.
     """
+    if dry_run:
+        return
+
+    # Try to change the permissions of the file or directory to delete them.
+    def onerror(func, path, exc_info):
+        if exc_info[0] == PermissionError:
+            # Parent directory must allow rwx
+            if not os.access(os.path.dirname(path), os.W_OK | os.R_OK | os.X_OK):
+                os.chmod(os.path.dirname(path), 0o0700)
+            if not os.access(path, os.W_OK | os.R_OK):
+                os.chmod(path, 0o0600)
```

Comment:
Make it less intrusive as for the parent directory.

Reply:
Not sure it's required to be done for the file. It's gonna get deleted.